### DISCUSSION TITLE
[Bug Report] Stash fails to install when database path is selected

### DISCUSSION THREAD

**Discussion by David-Maisonave** [stashapp/stash#4891] [Q&A]
**Describe the bug**
Tried to install Stash with the following path configured for the database:
_E:\Database\Stash_
I kept getting an error when trying to commit the installation. After clearing the path, the installation completed without an error.
Unfortunately, I don't know how to uninstall Stash so I can reproduce the problem, and provide more details.
If I can be given details on how to uninstall Stash, I could repeat this issue and give more details.

**To Reproduce**
1. Install the latest version of Stash via stash-win.exe
2. Set the following as the database path: **E:\Database\Stash**
3. Set the following as video content path: **B:\V**
4. Commit settings, and you'll get an error message giving you the option to click the "**BACK**" button.

**Expected behavior**
Expect installation to complete with the given database path.

**Stash Version: (from Settings -> About):**
**v0.25.1**
Build hash:
bf7cb78d
Build time:
2024-03-13 03:30:08

**Desktop (please complete the following information):**
 - OS: **Windows 11 Pro (x64)**
 - Browser **Firefox**
 - Version **124.0.2 (64-bit)**

**Comment by WithoutPants** [comment]
Stash gives you the option to create a system either in the current working directory (ie usually where the executable is) or in `$HOME/.stash` - which is usually `C:\Users\<user>\.stash`. The "uninstall" process involves deleting the `config.yml` file in the applicable location and any created directories (usually `generated`, `cache`, and `blobs`).

You may want to ensure that stash is not running - on Windows, it puts an icon in the notification area while its running (assuming its run by double clicking the exe and not run from a terminal).

**Comment by David-Maisonave** [comment]
My scan task has already been running over 24 hours, so I'm going to let it complete before simulating an uninstall.
Since I didn't want the database to be created under my **C:\Users\<user>** path, I created a path **E:\Stash** and then I created a Windows hardlink **C:\Users\<user>\.stash** which pointed to **E:\Stash**.

When the scan task completes, I'll delete the hardlink **C:\Users\<user>\.stash**, and hopefully Stash will see that as an uninstall.
I'll then see if I can reproduce the bug and if-so, I'll provide more details.

**Comment by David-Maisonave** [comment]
Sorry it took so long getting back to this.
I removed the ".stash" symbolic link from my user directory, and ran the latest version "v0.26.1" so-as to reproduce the issue.
Here's the screenshots.
<img width="373" alt="PathsSelectedOnInstall" src="https://github.com/stashapp/stash/assets/47364845/ae2e62fc-ab6d-4729-aea2-c6d07f577f8f">

![StashSetupWizard_Dialog](https://github.com/stashapp/stash/assets/47364845/6068e8b1-7725-4df9-ab80-c0bd2442e7f4)

And here's the error display:
![Oh_no!_Something_went_wrong!](https://github.com/stashapp/stash/assets/47364845/95ba9777-f57d-4e48-a0b1-3cf2332ff518)

More details to follow.

**Comment by David-Maisonave** [comment]
I figured out the problem. I failed to realized that the installation dialog was asking for a filename, and not a folder path. I was creating the folder path in advanced (**E:\Temp\Stash\db**), so when the installation program tried to create the file, it couldn't because there was already a directory with the same name in the same path.

Although this issue is a user error, I can see it happening to other users, because the other fields are folder paths, and it's rare for setup programs to ask for a filename. 
I recommend that the program check if the path is an existing directory, and if so, create the DB under that path.
Or at least give a detailed error message stating the path already exist as a directory.

**Comment by David-Maisonave** [comment]
FYI:
I just notice I had the same issue with the logging folder.
`Could not open 'Z:\[_Backups_]\Stash\Logging' for log output due to error: open Z:\[_Backups_]\Stash\Logging: is a directory`

I realize this is also user error, but the norm is for user to enter paths, and not file names.
